### ISSUE TITLE
interop: building js-libp2p v0.46 image fails

### ISSUE THREAD

**Issue by marten-seemann** [libp2p/test-plans#279]
Building locally (M1 MBP):
```
 => ERROR [4/6] RUN npm i && npm run build                                                                                                                                                                         407.5s
------
 > [4/6] RUN npm i && npm run build:
308.3 npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
308.6 npm WARN deprecated rollup-plugin-terser@7.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
308.6 npm WARN deprecated workbox-cacheable-response@6.6.0: workbox-background-sync@6.6.0
310.3 npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
312.3 npm WARN deprecated multibase@4.0.6: This module has been superseded by the multiformats module
312.6 npm WARN deprecated ipfs-core-utils@0.18.1: js-IPFS has been deprecated in favour of Helia - please see https://github.com/ipfs/js-ipfs/issues/4336 for details
312.7 npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
314.6 npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
315.3 npm WARN deprecated ipfs-core-types@0.14.1: js-IPFS has been deprecated in favour of Helia - please see https://github.com/ipfs/js-ipfs/issues/4336 for details
407.1 npm notice
407.1 npm notice New minor version of npm available! 9.6.7 -> 9.8.1
407.1 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.8.1>
407.1 npm notice Run `npm install -g npm@9.8.1` to update!
407.1 npm notice
407.1 npm ERR! code 2
407.1 npm ERR! path /app/node_modules/node-datachannel
407.1 npm ERR! command failed
407.1 npm ERR! command sh -c prebuild-install || (npm install --ignore-scripts --production=false && npm run _prebuild)
407.1 npm ERR! added 600 packages, removed 1 package, and audited 658 packages in 37s
407.1 npm ERR!
407.1 npm ERR! 68 packages are looking for funding
407.1 npm ERR!   run `npm fund` for details
407.1 npm ERR!
407.1 npm ERR! 10 vulnerabilities (6 moderate, 4 high)
407.1 npm ERR!
407.1 npm ERR! To address issues that do not require attention, run:
407.1 npm ERR!   npm audit fix
407.1 npm ERR!
407.1 npm ERR! To address all issues (including breaking changes), run:
407.1 npm ERR!   npm audit fix --force
407.1 npm ERR!
407.1 npm ERR! Run `npm audit` for details.
407.1 npm ERR!
407.1 npm ERR! > node-datachannel@0.4.3 _prebuild
407.1 npm ERR! > prebuild --backend cmake-js
407.1 npm ERR!
407.1 npm ERR! [
407.1 npm ERR!   '/usr/local/bin/node',
407.1 npm ERR!   '/app/node_modules/node-datachannel/node_modules/.bin/cmake-js',
407.1 npm ERR!   'rebuild',
407.1 npm ERR!   '--runtime-version=18.17.1',
407.1 npm ERR!   '--arch=arm64',
407.1 npm ERR!   '--runtime=node'
407.1 npm ERR! ]
407.1 npm ERR! prebuild-install WARN install libcrypto.so.1.1: cannot open shared object file: No such file or directory
407.1 npm ERR! npm WARN config production Use `--omit=dev` instead.
407.1 npm ERR! npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
407.1 npm ERR! npm WARN deprecated har-validator@5.1.5: this library is no longer supported
407.1 npm ERR! npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details.
407.1 npm ERR! npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
407.1 npm ERR! npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.
407.1 npm ERR! npm WARN deprecated tar@2.2.2: This version of tar is no longer supported, and will not receive security updates. Please upgrade asap.
407.1 npm ERR! prebuild info begin Prebuild version 11.0.4
407.1 npm ERR! prebuild info build Preparing to prebuild node-datachannel@0.4.3 for node 18.17.1 on linux-arm64 using cmake-js
407.1 npm ERR! info TOOL Using Unix Makefiles generator.
407.1 npm ERR! info DIST Downloading distribution files to: /root/.cmake-js/node-arm64/v18.17.1
407.1 npm ERR! http DIST 	- https://nodejs.org/dist/v18.17.1/SHASUMS256.txt
407.1 npm ERR! http DIST 	- https://nodejs.org/dist/v18.17.1/node-v18.17.1-headers.tar.gz
407.1 npm ERR! ERR! OMG CMake executable is not found. Please use your system's package manager to install it, or you can get installers from there: http://cmake.org.
407.1 npm ERR! prebuild ERR! build Error: Failed to build cmake with exit code 1
407.1 npm ERR! prebuild ERR! build     at ChildProcess.<anonymous> (/app/node_modules/node-datachannel/node_modules/prebuild/cmakebuild.js:32:19)
407.1 npm ERR! prebuild ERR! build     at ChildProcess.emit (node:events:514:28)
407.1 npm ERR! prebuild ERR! build     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
407.1
407.1 npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2023-08-25T12_12_25_260Z-debug-0.log
------
Dockerfile:6
--------------------
   4 |     WORKDIR /app
   5 |     COPY . .
   6 | >>> RUN npm i && npm run build
   7 |
   8 |     WORKDIR /app/interop
--------------------
ERROR: failed to solve: process "/bin/sh -c npm i && npm run build" did not complete successfully: exit code: 2
make: *** [image.json] Error 1

**Comment by marten-seemann** [comment]
And on my Ubuntu VM (using Parallels on my Mac) the build times out:
```
❯ BUILDKIT_PROGRESS=plain make
cd js-libp2p-b7e608998cc88860d9ec8a3ed7c03fdfb3eccb3b && docker build -t js-v0.46 -f ../Dockerfile .
#0 building with "default" instance using docker driver

#1 [internal] load .dockerignore
#1 transferring context: 70B done
#1 DONE 0.0s

#2 [internal] load build definition from Dockerfile
#2 transferring dockerfile: 438B done
#2 DONE 0.0s

#3 [internal] load metadata for docker.io/library/node:18
#3 DONE 1.6s

#4 [1/6] FROM docker.io/library/node:18@sha256:11e9c297fc51f6f65f7d0c7c8a8581e5721f2f16de43ceff1a199fd3ef609f95
#4 DONE 0.0s

#5 [internal] load build context
#5 transferring context: 73.71kB 0.0s done
#5 DONE 0.0s

#6 [2/6] WORKDIR /app
#6 CACHED

#7 [3/6] COPY . .
#7 CACHED

#8 [4/6] RUN npm i && npm run build
```
The last step took more than an hour without outputting anything.

**Comment by marten-seemann** [comment]
On my Linode cloud server (x86_64):
```
=> ERROR [4/6] RUN npm i && npm run build                                                                                                                                                                         212.7s
------
 > [4/6] RUN npm i && npm run build:
82.40 npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
82.43 npm WARN deprecated rollup-plugin-terser@7.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
83.12 npm WARN deprecated workbox-cacheable-response@6.6.0: workbox-background-sync@6.6.0
83.32 npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
84.60 npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
84.84 npm WARN deprecated multibase@4.0.6: This module has been superseded by the multiformats module
87.23 npm WARN deprecated ipfs-core-utils@0.18.1: js-IPFS has been deprecated in favour of Helia - please see https://github.com/ipfs/js-ipfs/issues/4336 for details
89.05 npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
97.42 npm WARN deprecated ipfs-core-types@0.14.1: js-IPFS has been deprecated in favour of Helia - please see https://github.com/ipfs/js-ipfs/issues/4336 for details
112.9
112.9 added 2960 packages, and audited 3305 packages in 2m
112.9
112.9 478 packages are looking for funding
112.9   run `npm fund` for details
113.0
113.0 7 vulnerabilities (1 moderate, 6 high)
113.0
113.0 To address issues that do not require attention, run:
113.0   npm audit fix
113.0
113.0 To address all issues (including breaking changes), run:
113.0   npm audit fix --force
113.0
113.0 Run `npm audit` for details.
113.0 npm notice
113.0 npm notice New minor version of npm available! 9.6.7 -> 9.8.1
113.0 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.8.1>
113.0 npm notice Run `npm install -g npm@9.8.1` to update!
113.0 npm notice
113.3
113.3 > js-libp2p-monorepo@1.0.0 build
113.3 > aegir run build
113.3
121.8
121.8 @libp2p/interface > npm run build
122.3
122.3 > @libp2p/interface@0.1.1 build
122.3 > aegir build
122.3
123.6 [03:46:39] tsc [started]
126.3 [03:46:42] tsc [completed]
126.3 [03:46:42] esbuild [started]
126.4 [03:46:42] esbuild [completed]
126.4
126.4 @libp2p/peer-id > npm run build
126.8
126.8 > @libp2p/peer-id@3.0.1 build
126.8 > aegir build
126.8
128.1 [03:46:44] tsc [started]
130.5 [03:46:46] tsc [completed]
130.5 [03:46:46] esbuild [started]
130.5 [03:46:46] esbuild [completed]
130.6
130.6 @libp2p/logger > npm run build
130.9
130.9 > @libp2p/logger@3.0.1 build
130.9 > aegir build
130.9
132.2 [03:46:48] tsc [started]
134.5 [03:46:50] tsc [completed]
134.5 [03:46:50] esbuild [started]
134.6 [03:46:50] esbuild [completed]
134.6
134.6 @libp2p/crypto > npm run build
134.9
134.9 > @libp2p/crypto@2.0.2 build
134.9 > aegir build
134.9
136.2 [03:46:52] tsc [started]
139.6 [03:46:55] tsc [completed]
139.6 [03:46:55] esbuild [started]
139.7 [03:46:55] esbuild [completed]
139.7
139.7 @libp2p/peer-id-factory > npm run build
140.0
140.0 > @libp2p/peer-id-factory@3.0.2 build
140.0 > aegir build
140.0
141.3 [03:46:57] tsc [started]
143.6 [03:46:59] tsc [completed]
143.6 [03:46:59] esbuild [started]
143.7 [03:46:59] esbuild [completed]
143.7
143.7 @libp2p/peer-collections > npm run build
144.1
144.1 > @libp2p/peer-collections@4.0.2 build
144.1 > aegir build
144.1
145.4 [03:47:01] tsc [started]
148.0 [03:47:04] tsc [completed]
148.0 [03:47:04] esbuild [started]
148.1 [03:47:04] esbuild [completed]
148.1
148.1 @libp2p/interface-internal > npm run build
148.5
148.5 > @libp2p/interface-internal@0.1.2 build
148.5 > aegir build
148.5
149.8 [03:47:05] tsc [started]
151.9 [03:47:07] tsc [completed]
151.9 [03:47:07] esbuild [started]
151.9 [03:47:07] esbuild [completed]
151.9
151.9 @libp2p/multistream-select > npm run build
152.2
152.2 > @libp2p/multistream-select@4.0.1 build
152.2 > aegir build
152.2
153.5 [03:47:09] tsc [started]
156.3 [03:47:12] tsc [completed]
156.3 [03:47:12] esbuild [started]
156.3 [03:47:12] esbuild [completed]
156.3
156.3 @libp2p/interface-compliance-tests > npm run build
156.7
156.7 > @libp2p/interface-compliance-tests@4.0.2 build
156.7 > aegir build
156.7
158.0 [03:47:14] tsc [started]
161.5 [03:47:17] tsc [completed]
161.5 [03:47:17] esbuild [started]
161.5 [03:47:17] esbuild [completed]
161.6
161.6 @libp2p/mplex > npm run build
161.9
161.9 > @libp2p/mplex@9.0.2 build
161.9 > aegir build
161.9
163.2 [03:47:19] tsc [started]
166.0 [03:47:22] tsc [completed]
166.0 [03:47:22] esbuild [started]
166.0 [03:47:22] esbuild [completed]
166.0
166.0 @libp2p/prometheus-metrics > npm run build
166.4
166.4 > @libp2p/prometheus-metrics@2.0.2 build
166.4 > aegir build
166.4
167.6 [03:47:23] tsc [started]
170.2 [03:47:26] tsc [completed]
170.2
170.2 @libp2p/utils > npm run build
170.6
170.6 > @libp2p/utils@4.0.1 build
170.6 > aegir build
170.6
171.9 [03:47:27] tsc [started]
174.2 [03:47:30] tsc [completed]
174.2 [03:47:30] esbuild [started]
174.2 [03:47:30] esbuild [completed]
174.2
174.2 @libp2p/tcp > npm run build
174.5
174.5 > @libp2p/tcp@8.0.2 build
174.5 > aegir build
174.5
175.8 [03:47:31] tsc [started]
178.6 [03:47:34] tsc [completed]
178.6 [03:47:34] esbuild [started]
178.7 [03:47:34] esbuild [completed]
178.7
178.7 @libp2p/keychain > npm run build
179.1
179.1 > @libp2p/keychain@3.0.2 build
179.1 > aegir build
179.1
180.3 [03:47:36] tsc [started]
182.8 [03:47:38] tsc [completed]
182.8 [03:47:38] esbuild [started]
182.9 [03:47:38] esbuild [completed]
182.9
182.9 @libp2p/peer-record > npm run build
183.3
183.3 > @libp2p/peer-record@6.0.2 build
183.3 > aegir build
183.3
184.5 [03:47:40] tsc [started]
186.9 [03:47:42] tsc [completed]
186.9 [03:47:42] esbuild [started]
187.0 [03:47:43] esbuild [completed]
187.0
187.0 @libp2p/peer-store > npm run build
187.4
187.4 > @libp2p/peer-store@9.0.2 build
187.4 > aegir build
187.4
188.6 [03:47:44] tsc [started]
191.8 [03:47:47] tsc [completed]
191.8 [03:47:47] esbuild [started]
192.0 [03:47:48] esbuild [completed]
192.0
192.0 @libp2p/bootstrap > npm run build
192.4
192.4 > @libp2p/bootstrap@9.0.2 build
192.4 > aegir build
192.4
193.7 [03:47:49] tsc [started]
196.0 [03:47:52] tsc [completed]
196.0 [03:47:52] esbuild [started]
196.0 [03:47:52] esbuild [completed]
196.0
196.0 @libp2p/pubsub > npm run build
196.4
196.4 > @libp2p/pubsub@8.0.3 build
196.4 > aegir build
196.4
197.7 [03:47:53] tsc [started]
200.9 [03:47:57] tsc [completed]
200.9 [03:47:57] esbuild [started]
201.0 [03:47:57] esbuild [completed]
201.1
201.1 @libp2p/floodsub > npm run build
201.4
201.4 > @libp2p/floodsub@8.0.3 build
201.4 > aegir build
201.4
202.7 [03:47:58] tsc [started]
205.3 [03:48:01] tsc [completed]
205.3 [03:48:01] esbuild [started]
205.4 [03:48:01] esbuild [completed]
205.4
205.4 @libp2p/kad-dht > npm run build
205.8
205.8 > @libp2p/kad-dht@10.0.2 build
205.8 > aegir build
205.8
207.1 [03:48:03] tsc [started]
212.3 src/providers.ts(100,21): error TS2769: No overload matches this call.
212.3   Overload 1 of 3, '(intervalId: string | number | Timeout | undefined): void', gave the following error.
212.3     Argument of type 'Timer' is not assignable to parameter of type 'string | number | Timeout | undefined'.
212.3       Property '[Symbol.dispose]' is missing in type 'Timer' but required in type 'Timeout'.
212.3   Overload 2 of 3, '(id: number | undefined): void', gave the following error.
212.3     Argument of type 'Timer' is not assignable to parameter of type 'number'.
212.3   Overload 3 of 3, '(id: number | undefined): void', gave the following error.
212.3     Argument of type 'Timer' is not assignable to parameter of type 'number'.
212.3 src/query-self.ts(74,18): error TS2769: No overload matches this call.
212.3   Overload 1 of 3, '(timeoutId: string | number | Timeout | undefined): void', gave the following error.
212.3     Argument of type 'Timer | undefined' is not assignable to parameter of type 'string | number | Timeout | undefined'.
212.3       Type 'Timer' is not assignable to type 'string | number | Timeout | undefined'.
212.3   Overload 2 of 3, '(id: number | undefined): void', gave the following error.
212.3     Argument of type 'Timer | undefined' is not assignable to parameter of type 'number | undefined'.
212.3       Type 'Timer' is not assignable to type 'number'.
212.3   Overload 3 of 3, '(id: number | undefined): void', gave the following error.
212.3     Argument of type 'Timer | undefined' is not assignable to parameter of type 'number | undefined'.
212.3 src/query-self.ts(87,20): error TS2769: No overload matches this call.
212.3   Overload 1 of 3, '(timeoutId: string | number | Timeout | undefined): void', gave the following error.
212.3     Argument of type 'Timer' is not assignable to parameter of type 'string | number | Timeout | undefined'.
212.3   Overload 2 of 3, '(id: number | undefined): void', gave the following error.
212.3     Argument of type 'Timer' is not assignable to parameter of type 'number'.
212.3   Overload 3 of 3, '(id: number | undefined): void', gave the following error.
212.3     Argument of type 'Timer' is not assignable to parameter of type 'number'.
212.3 src/routing-table/refresh.ts(59,20): error TS2769: No overload matches this call.
212.3   Overload 1 of 3, '(timeoutId: string | number | Timeout | undefined): void', gave the following error.
212.3     Argument of type 'Timer' is not assignable to parameter of type 'string | number | Timeout | undefined'.
212.3   Overload 2 of 3, '(id: number | undefined): void', gave the following error.
212.3     Argument of type 'Timer' is not assignable to parameter of type 'number'.
212.3   Overload 3 of 3, '(id: number | undefined): void', gave the following error.
212.3     Argument of type 'Timer' is not assignable to parameter of type 'number'.
212.4 [03:48:08] tsc [failed]
212.4 [03:48:08] → Command failed with exit code 1: tsc
212.4 Command failed with exit code 1: tsc
212.4 npm ERR! Lifecycle script `build` failed with error:
212.4 npm ERR! Error: command failed
212.4 npm ERR!   in workspace: @libp2p/kad-dht@10.0.2
212.4 npm ERR!   at location: /app/packages/kad-dht
212.4 Command failed with exit code 1: npm run build
------
Dockerfile:6
--------------------
   4 |     WORKDIR /app
   5 |     COPY . .
   6 | >>> RUN npm i && npm run build
   7 |
   8 |     WORKDIR /app/interop
--------------------
ERROR: failed to solve: process "/bin/sh -c npm i && npm run build" did not complete successfully: exit code: 1
make: *** [Makefile:22: image.json] Error 1
 ~/src/go/src/github.com/libp2p/test-plans/multidim-interop/impl/js/v0.46 | master                                                                                                              4m 9s | root@linode-us
❯ make chromium-image.json
cd js-libp2p-b7e608998cc88860d9ec8a3ed7c03fdfb3eccb3b && docker build -t js-v0.46 -f ../Dockerfile .
[+] Building 209.3s (8/10)                                                                                                                                                                                 docker:default
 => [internal] load build definition from Dockerfile                                                                                                                                                                 0.0s
 => => transferring dockerfile: 438B                                                                                                                                                                                 0.0s
 => [internal] load .dockerignore                                                                                                                                                                                    0.0s
 => => transferring context: 70B                                                                                                                                                                                     0.0s
 => [internal] load metadata for docker.io/library/node:18                                                                                                                                                           0.3s
 => [1/6] FROM docker.io/library/node:18@sha256:11e9c297fc51f6f65f7d0c7c8a8581e5721f2f16de43ceff1a199fd3ef609f95                                                                                                     0.0s
 => [internal] load build context                                                                                                                                                                                    0.1s
 => => transferring context: 73.71kB                                                                                                                                                                                 0.1s
 => CACHED [2/6] WORKDIR /app                                                                                                                                                                                        0.0s
 => CACHED [3/6] COPY . .                                                                                                                                                                                            0.0s
 => ERROR [4/6] RUN npm i && npm run build                                                                                                                                                                         208.9s
------
 > [4/6] RUN npm i && npm run build:
84.90 npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
84.92 npm WARN deprecated rollup-plugin-terser@7.0.2: This package has been deprecated and is no longer maintained. Please use @rollup/plugin-terser
85.42 npm WARN deprecated workbox-cacheable-response@6.6.0: workbox-background-sync@6.6.0
85.59 npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
86.56 npm WARN deprecated w3c-hr-time@1.0.2: Use your platform's native performance.now() and performance.timeOrigin.
87.20 npm WARN deprecated multibase@4.0.6: This module has been superseded by the multiformats module
88.36 npm WARN deprecated ipfs-core-utils@0.18.1: js-IPFS has been deprecated in favour of Helia - please see https://github.com/ipfs/js-ipfs/issues/4336 for details
89.04 npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
95.91 npm WARN deprecated ipfs-core-types@0.14.1: js-IPFS has been deprecated in favour of Helia - please see https://github.com/ipfs/js-ipfs/issues/4336 for details
110.9
110.9 added 2960 packages, and audited 3305 packages in 2m
110.9
110.9 478 packages are looking for funding
110.9   run `npm fund` for details
110.9
110.9 7 vulnerabilities (1 moderate, 6 high)
110.9
110.9 To address issues that do not require attention, run:
110.9   npm audit fix
110.9
110.9 To address all issues (including breaking changes), run:
110.9   npm audit fix --force
110.9
110.9 Run `npm audit` for details.
110.9 npm notice
110.9 npm notice New minor version of npm available! 9.6.7 -> 9.8.1
110.9 npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.8.1>
110.9 npm notice Run `npm install -g npm@9.8.1` to update!
110.9 npm notice
111.2
111.2 > js-libp2p-monorepo@1.0.0 build
111.2 > aegir run build
111.2
119.9
119.9 @libp2p/interface > npm run build
120.4
120.4 > @libp2p/interface@0.1.1 build
120.4 > aegir build
120.4
121.7 [03:52:47] tsc [started]
124.3 [03:52:49] tsc [completed]
124.3 [03:52:49] esbuild [started]
124.4 [03:52:49] esbuild [completed]
124.4
124.4 @libp2p/peer-id > npm run build
124.7
124.7 > @libp2p/peer-id@3.0.1 build
124.7 > aegir build
124.7
126.0 [03:52:51] tsc [started]
128.3 [03:52:53] tsc [completed]
128.3 [03:52:53] esbuild [started]
128.4 [03:52:53] esbuild [completed]
128.4
128.4 @libp2p/logger > npm run build
128.7
128.7 > @libp2p/logger@3.0.1 build
128.7 > aegir build
128.7
130.0 [03:52:55] tsc [started]
132.2 [03:52:57] tsc [completed]
132.2 [03:52:57] esbuild [started]
132.2 [03:52:57] esbuild [completed]
132.2
132.2 @libp2p/crypto > npm run build
132.6
132.6 > @libp2p/crypto@2.0.2 build
132.6 > aegir build
132.6
133.9 [03:52:59] tsc [started]
137.5 [03:53:02] tsc [completed]
137.5 [03:53:02] esbuild [started]
137.6 [03:53:03] esbuild [completed]
137.6
137.6 @libp2p/peer-id-factory > npm run build
137.9
137.9 > @libp2p/peer-id-factory@3.0.2 build
137.9 > aegir build
137.9
139.2 [03:53:04] tsc [started]
141.6 [03:53:06] tsc [completed]
141.6 [03:53:06] esbuild [started]
141.7 [03:53:07] esbuild [completed]
141.7
141.7 @libp2p/peer-collections > npm run build
142.0
142.0 > @libp2p/peer-collections@4.0.2 build
142.0 > aegir build
142.0
143.3 [03:53:08] tsc [started]
145.6 [03:53:11] tsc [completed]
145.6 [03:53:11] esbuild [started]
145.6 [03:53:11] esbuild [completed]
145.6
145.6 @libp2p/interface-internal > npm run build
146.0
146.0 > @libp2p/interface-internal@0.1.2 build
146.0 > aegir build
146.0
147.3 [03:53:12] tsc [started]
149.4 [03:53:14] tsc [completed]
149.4 [03:53:14] esbuild [started]
149.4 [03:53:14] esbuild [completed]
149.4
149.4 @libp2p/multistream-select > npm run build
149.7
149.7 > @libp2p/multistream-select@4.0.1 build
149.7 > aegir build
149.7
151.0 [03:53:16] tsc [started]
153.6 [03:53:19] tsc [completed]
153.6 [03:53:19] esbuild [started]
153.6 [03:53:19] esbuild [completed]
153.7
153.7 @libp2p/interface-compliance-tests > npm run build
154.0
154.0 > @libp2p/interface-compliance-tests@4.0.2 build
154.0 > aegir build
154.0
155.2 [03:53:20] tsc [started]
158.8 [03:53:24] tsc [completed]
158.8 [03:53:24] esbuild [started]
158.8 [03:53:24] esbuild [completed]
158.8
158.8 @libp2p/mplex > npm run build
159.2
159.2 > @libp2p/mplex@9.0.2 build
159.2 > aegir build
159.2
160.4 [03:53:25] tsc [started]
163.3 [03:53:28] tsc [completed]
163.3 [03:53:28] esbuild [started]
163.4 [03:53:28] esbuild [completed]
163.4
163.4 @libp2p/prometheus-metrics > npm run build
163.8
163.8 > @libp2p/prometheus-metrics@2.0.2 build
163.8 > aegir build
163.8
165.1 [03:53:30] tsc [started]
167.8 [03:53:33] tsc [completed]
167.9
167.9 @libp2p/utils > npm run build
168.2
168.2 > @libp2p/utils@4.0.1 build
168.2 > aegir build
168.2
169.4 [03:53:34] tsc [started]
171.8 [03:53:37] tsc [completed]
171.8 [03:53:37] esbuild [started]
171.8 [03:53:37] esbuild [completed]
171.8
171.8 @libp2p/tcp > npm run build
172.1
172.1 > @libp2p/tcp@8.0.2 build
172.1 > aegir build
172.1
173.4 [03:53:38] tsc [started]
176.1 [03:53:41] tsc [completed]
176.1 [03:53:41] esbuild [started]
176.2 [03:53:41] esbuild [completed]
176.2
176.2 @libp2p/keychain > npm run build
176.5
176.5 > @libp2p/keychain@3.0.2 build
176.5 > aegir build
176.5
177.8 [03:53:43] tsc [started]
180.3 [03:53:45] tsc [completed]
180.3 [03:53:45] esbuild [started]
180.4 [03:53:45] esbuild [completed]
180.4
180.4 @libp2p/peer-record > npm run build
180.8
180.8 > @libp2p/peer-record@6.0.2 build
180.8 > aegir build
180.8
182.0 [03:53:47] tsc [started]
184.4 [03:53:49] tsc [completed]
184.4 [03:53:49] esbuild [started]
184.5 [03:53:49] esbuild [completed]
184.5
184.5 @libp2p/peer-store > npm run build
184.9
184.9 > @libp2p/peer-store@9.0.2 build
184.9 > aegir build
184.9
186.2 [03:53:51] tsc [started]
189.1 [03:53:54] tsc [completed]
189.1 [03:53:54] esbuild [started]
189.2 [03:53:54] esbuild [completed]
189.2
189.2 @libp2p/bootstrap > npm run build
189.5
189.5 > @libp2p/bootstrap@9.0.2 build
189.5 > aegir build
189.5
190.8 [03:53:56] tsc [started]
193.1 [03:53:58] tsc [completed]
193.1 [03:53:58] esbuild [started]
193.1 [03:53:58] esbuild [completed]
193.1
193.1 @libp2p/pubsub > npm run build
193.5
193.5 > @libp2p/pubsub@8.0.3 build
193.5 > aegir build
193.5
194.7 [03:54:00] tsc [started]
197.8 [03:54:03] tsc [completed]
197.9 [03:54:03] esbuild [started]
197.9 [03:54:03] esbuild [completed]
198.0
198.0 @libp2p/floodsub > npm run build
198.3
198.3 > @libp2p/floodsub@8.0.3 build
198.3 > aegir build
198.3
199.6 [03:54:05] tsc [started]
202.1 [03:54:07] tsc [completed]
202.1 [03:54:07] esbuild [started]
202.2 [03:54:07] esbuild [completed]
202.2
202.2 @libp2p/kad-dht > npm run build
202.6
202.6 > @libp2p/kad-dht@10.0.2 build
202.6 > aegir build
202.6
203.8 [03:54:09] tsc [started]
208.5 src/providers.ts(100,21): error TS2769: No overload matches this call.
208.5   Overload 1 of 3, '(intervalId: string | number | Timeout | undefined): void', gave the following error.
208.5     Argument of type 'Timer' is not assignable to parameter of type 'string | number | Timeout | undefined'.
208.5       Property '[Symbol.dispose]' is missing in type 'Timer' but required in type 'Timeout'.
208.5   Overload 2 of 3, '(id: number | undefined): void', gave the following error.
208.5     Argument of type 'Timer' is not assignable to parameter of type 'number'.
208.5   Overload 3 of 3, '(id: number | undefined): void', gave the following error.
208.5     Argument of type 'Timer' is not assignable to parameter of type 'number'.
208.5 src/query-self.ts(74,18): error TS2769: No overload matches this call.
208.5   Overload 1 of 3, '(timeoutId: string | number | Timeout | undefined): void', gave the following error.
208.5     Argument of type 'Timer | undefined' is not assignable to parameter of type 'string | number | Timeout | undefined'.
208.5       Type 'Timer' is not assignable to type 'string | number | Timeout | undefined'.
208.5   Overload 2 of 3, '(id: number | undefined): void', gave the following error.
208.5     Argument of type 'Timer | undefined' is not assignable to parameter of type 'number | undefined'.
208.5       Type 'Timer' is not assignable to type 'number'.
208.5   Overload 3 of 3, '(id: number | undefined): void', gave the following error.
208.5     Argument of type 'Timer | undefined' is not assignable to parameter of type 'number | undefined'.
208.5 src/query-self.ts(87,20): error TS2769: No overload matches this call.
208.5   Overload 1 of 3, '(timeoutId: string | number | Timeout | undefined): void', gave the following error.
208.5     Argument of type 'Timer' is not assignable to parameter of type 'string | number | Timeout | undefined'.
208.5   Overload 2 of 3, '(id: number | undefined): void', gave the following error.
208.5     Argument of type 'Timer' is not assignable to parameter of type 'number'.
208.5   Overload 3 of 3, '(id: number | undefined): void', gave the following error.
208.5     Argument of type 'Timer' is not assignable to parameter of type 'number'.
208.5 src/routing-table/refresh.ts(59,20): error TS2769: No overload matches this call.
208.5   Overload 1 of 3, '(timeoutId: string | number | Timeout | undefined): void', gave the following error.
208.5     Argument of type 'Timer' is not assignable to parameter of type 'string | number | Timeout | undefined'.
208.5   Overload 2 of 3, '(id: number | undefined): void', gave the following error.
208.5     Argument of type 'Timer' is not assignable to parameter of type 'number'.
208.5   Overload 3 of 3, '(id: number | undefined): void', gave the following error.
208.5     Argument of type 'Timer' is not assignable to parameter of type 'number'.
208.5 [03:54:13] tsc [failed]
208.5 [03:54:13] → Command failed with exit code 1: tsc
208.6 Command failed with exit code 1: tsc
208.6 npm ERR! Lifecycle script `build` failed with error:
208.6 npm ERR! Error: command failed
208.6 npm ERR!   in workspace: @libp2p/kad-dht@10.0.2
208.6 npm ERR!   at location: /app/packages/kad-dht
208.6 Command failed with exit code 1: npm run build
------
Dockerfile:6
--------------------
   4 |     WORKDIR /app
   5 |     COPY . .
   6 | >>> RUN npm i && npm run build
   7 |
   8 |     WORKDIR /app/interop
--------------------
ERROR: failed to solve: process "/bin/sh -c npm i && npm run build" did not complete successfully: exit code: 1
make: *** [Makefile:22: image.json] Error 1
```

**Comment by MarcoPolo** [comment]
> I think going forward it may be more stable to release the js multidim interop tests in an npm package (similar to [the js-perf package](https://www.npmjs.com/package/@libp2p/perf)) @achingbrain @MarcoPolo

Ultimately we need to pin the dependencies. I took a stab at that in #286 .

Installing cmake in the container fixes the issue on arm64 linux. I'm guessing some prebuilt binary was unavailable.

**Comment by p-shahi** [comment]
Closed by #288
